### DISCUSSION TITLE
Using remote control in tmux through ssh.

### DISCUSSION THREAD

**Discussion by NikoKS** [kovidgoyal/kitty#3283] [Q&A] [answered]
Hi, I'm trying to use kitty remote control on an ssh'ed server running tmux. Somehow this doesn't seem to be possible.

for example, kitty @ ls works through ssh, works inside tmux, but not inside tmux over ssh.

What I've tried:
1. try running kitty @ ls. keep gives me time out error similar to [2338](https://github.com/kovidgoyal/kitty/issues/2338)
2. set to listen on tcp:localhost:12345 socket and do kitty @ --to=tcp:localhost:12345. Keep getting connection refused

Does anyone ever succeed in trying this?

**Comment by kovidgoyal** [comment, marked as answer]
tmux will be filtering out the escape codes kitty uses for remote control, so you cant. As always with tmux issues: https://sw.kovidgoyal.net/kitty/faq.html#i-am-using-tmux-and-have-a-problem

If you really want to do it still, you could tell kitty to listen on a tcp port, forward it to the remote host via ssh and use --to tcp:forwarded-port

**Reply by NikoKS** [reply]
Hi Kovid. Thank you very much for your reply.

Unfortunately, I still do need tmux's remote persistence since I'm almost always working on remote system. (Really wish kitty would be able to do this in the future 😉)

Just test using ssh -R option to forward the port and it works!

May I know why it's able to work without port forwarding when not inside tmux?
Also, are you aware of any fatal security flaws doing port forwarding like this?
